homeassistant/components/withings/sensor.py

```diff
@@ -0,0 +1,1057 @@
+                          config_filename: str,
+                          creds) -> None:
+    """Handle calls from  when the nokia api refreshes credentials."""
+    _LOGGER.debug('async_credentials_refreshed')
+    hass.add_job(write_credentials_to_file, hass, config_filename, creds)
+
+
+class WithingsConfiguring:
+    """Hold information used while configuring this component."""
+
+    request_id = None
+
+    def __init__(self,
+                 hass,
+                 config,
+                 add_entities,
+                 slug,
+                 config_filename,
+                 oauth_initialize_callback,
+                 auth_client):
+        """Constructor."""
+        self.hass = hass
+        self.config = config
+        self.add_entities = add_entities
+        self.slug = slug
+        self.config_filename = config_filename
+        self.oauth_initialize_callback = oauth_initialize_callback
+        self.auth_client = auth_client
+
+
+async def async_initialize(
+        configuring: WithingsConfiguring,
+        creds) -> WithingsDataManager:
+    """Initialize the Withings data manager object from the created session."""
+    import nokia
+
+    _LOGGER.debug('async_initialize')
+    api = nokia.NokiaApi(
+        creds,
+
+        refresh_cb=(lambda token: credentials_refreshed(
+            configuring.hass,
+            configuring.config_filename,
+            api.credentials
+        ))
+    )
+
+    _LOGGER.debug('Saving the token data..')
+    credentials_refreshed(
+        configuring.hass,
+        configuring.config_filename,
+        api.credentials
+    )
+
+    _LOGGER.debug(
+        'Creating withings data manager for slug: %s',
+        configuring.slug
+    )
+    data_manager = WithingsDataManager(
+        configuring.slug,
+        api
+    )
+
+    _LOGGER.debug('Attempting to refresh token.')
+    await data_manager.async_refresh_token()
+
+    _LOGGER.debug('Creating entities.')
+    entities = []
+    measurements = configuring.config.get(CONF_MEASUREMENTS)
+    for measurement in measurements:
+        _LOGGER.debug('Creating entity for %s', measurement)
+
+        attribute = WITHINGS_MEASUREMENTS_MAP[measurement]
+
+        entity = WithingsHealthSensor(data_manager, attribute)
+
+        entities.append(entity)
+
+    _LOGGER.debug('Adding entities.')
+    configuring.add_entities(entities)
+
+    return data_manager
+
+
+async def async_oauth_initialize_callback(
+        code: str,
+        configuring: WithingsConfiguring) -> None:
+    """Call after OAuth2 response is returned."""
+    _LOGGER.debug('async_oauth_initialize_callback')
+
+    _LOGGER.debug('Requesting credentials with code: %s.', code)
+    creds = configuring.auth_client.get_credentials(code)
+
+    _LOGGER.debug('Initializing data.')
+    await async_initialize(configuring, creds)
+
+    _LOGGER.debug('Finishing request.')
+    configuring.hass.components.configurator.async_request_done(
```

Comment:
We should no longer use the configurator. We should use a config flow instead.

Reply:
Are there any good examples available for using a ConfigFlow? I based this platform on other components I found in source but didn't see anything about config flow.

Reply:
A good start would be the architecture docs. https://developers.home-assistant.io/docs/en/config_entries_config_flow_handler.html

Reply:
Look at nest and point components.